pvanalytics/quality/irradiance.py

```diff
@@ -471,3 +471,185 @@ def daily_insolation_limits(irrad, clearsky, daily_min=0.4, daily_max=1.25):
         lower_bound=daily_min
     )
     return good_days.reindex(irrad.index, method='pad', fill_value=False)
+
+
+def _fill_nighttime(component, component_sum_df, fill_night_value,
+                    solar_zenith, zenith_limit):
+    # This function is used to fill in nighttime values for the computed
+    # irradiance time series (GHI, DHI, DNI).
+    # Set the series based on the component.
+    if component == 'GHI':
+        series = component_sum_df['ghi']
+    elif component == 'DHI':
+        series = component_sum_df['dhi']
+    else:
+        series = component_sum_df['dni']
+    # Logic for filling in nighttime values for a
+    # component sum series.
+    # Find the locations where the sun is below the sza limit.
+    mask = (zenith_limit <= solar_zenith)
+    if isinstance(fill_night_value, float) | isinstance(fill_night_value, int):
+        # Replace the nighttime values with a fill value--this can be np.nan,
+        # which is a float
+        series[mask] = fill_night_value
+    elif fill_night_value == 'equation':
+        # Use the nighttime equation GHI = 0 + DHI, which translates as:
+        # GHI_Calc (at night) = DHI_measured
+        # DHI_Calc (at night) = GHI_measured
+        # DNI_Calc (at night) = 0
+        if component == 'GHI':
+            series[mask] = component_sum_df['dhi'][mask]
+        elif component == 'DHI':
+            series[mask] = component_sum_df['ghi'][mask]
+        else:
+            series[mask] = 0
+    elif fill_night_value is None:
+        pass
+    else:
+        raise ValueError("The fill_night_value variable must be None,"
+                         " float or int, or 'equation'. Please change "
+                         "the variable fill_night_value value.")
+    return series
+
+
+def _complete_irradiance(solar_zenith,
+                         ghi=None,
+                         dhi=None,
+                         dni=None,
+                         dni_clear=None):
+    """
+    TODO: This method exists in the pvlib-python library. Once a new PVLib
+    release or pre-release is cut, this private function can be deleted and
+    the associated PVLib function can be directly leveraged.
+
+    Use the component sum equations to calculate the missing series, using
+    the other available time series. One of the three parameters (ghi, dhi,
+    dni) is passed as None, and the other associated series passed are used to
+    calculate the missing series value.
+    The "component sum" or "closure" equation relates the three
+    primary irradiance components as follows:
+    .. math::
+       GHI = DHI + DNI * cos(theta_z)
```

Comment:
```suggestion
    primary irradiance components as follows:

    .. math::
       GHI = DHI + DNI * cos(\theta_z)
```

Reply:
The backslash here might need the entire docstring to be prefixed with `r`, same as https://github.com/pvlib/pvlib-python/pull/1567#discussion_r996135976.  As a side note, I think it is better to have `\cos` instead of `cos` (renders as roman versus italics).